src/ponyfills/minipromise.js

```diff
@@ -0,0 +1,67 @@
+const invoker = function(args, fn) {
+  fn.call(this, args);
+};
+
+export default class MiniPromise {
```

Comment:
Was this class hand-rolled? If so, I would highly recommend using code from an outside, well-tested library. Reinventing promises is fraught with difficulty, and we have no tests for this.

The promise shim that I use in `execution-plan` was extracted from [promise-polyfill](https://www.npmjs.com/package/promise-polyfill), and I have tests behind it. It also implements `Promise.resolve()`, `Promise.reject()`, `Promise.all()` and `Promise.race()`.

https://github.com/SentientTechnologies/execution-plan/blob/master/src/external-shims/promise.shim.ts
https://github.com/SentientTechnologies/execution-plan/blob/master/src/external-shims/promise.shim.test.ts

Reply:
This library isn't meant the replace the full promises functionality, and this library doesn't use any of the invocation side of the promises api (Promise.resolve(), Promise.reject(), Promise.all() and Promise.race().). This library only duck types the signatures where a response is needed to allow support for the use of await. Shimming the full Promise implementation is neither needed nor useful.

Reply:
I do believe Frazer plans on using `Promise.all()` in his asset renderer code.

Nevertheless the question still stands about whether this is hand-rolled or not. Promises are a treacherous place to have NIH syndrome.